types/react-native/index.d.ts

```diff
@@ -469,6 +471,84 @@ export interface Insets {
     right?: number;
 }
 
+export interface PressableStateCallbackType {
+    pressed: boolean;
+}
+
+export interface PressableProps extends AccessibilityProps, Omit<ViewProps, 'style' | 'hitSlop'> {
+    /**
+     * Called when a single tap gesture is detected.
+     */
+    onPress?: (event: GestureResponderEvent) => void;
```

Comment:
[The upstream type](https://github.com/facebook/react-native/blob/4f897336b6ce6c640a65f86b65c6d386ea55b944/Libraries/Components/Pressable/Pressable.js#L97) is a ‘maybe’ type, meaning it can be `null` too. Same applies to most (all?) other props below.

```suggestion
    onPress?: null | (event: GestureResponderEvent) => void;
```